### PR TITLE
feat(infra): Support checking proxyAdmin addresses in check-deploy

### PR DESCRIPTION
**Pull request by Mo-Hussain**
### Description

- Enable checking the proxyAdmin address for proxied contracts (Collateral, Synthetic) against what is set in config. This check assumes we use one proxyadmin contract per chain for (Mailbox, Collateral, Synthetic)

Example usage:

#### Ancient8

```
$yarn tsx ./scripts/check-deploy.ts -e mainnet3 -m warp -c ancient8

Deploying staticAggregationIsm to ethereum 
Deploying messageIdMultisigIsm to ethereum 
Recovered 2 of 3 address set on ethereum: 0xD32513263Eb3190c4DA126fe418864904D9F4A32
Deploying merkleRootMultisigIsm to ethereum 
Recovered 2 of 3 address set on ethereum: 0x054B75b89DA7a428C2BF1f6b5df33944167B785C
Recovered 1 of 2 address set on ethereum: 0x9c764dF6bdd6B716CfadEAa348E1dB146694F67F
┌─────────┬────────────┬────────┬─────────────┬─────────────────────────┬─────────┬──────────────────────────────────────────────┬──────────────────────────────────────────────┐
│ (index) │ chain      │ remote │ name        │ type                    │ subType │ actual                                       │ expected                                     │
├─────────┼────────────┼────────┼─────────────┼─────────────────────────┼─────────┼──────────────────────────────────────────────┼──────────────────────────────────────────────┤
│ 0       │ 'ancient8' │        │             │ 'TokenSymbolMismatch'   │         │ 'USDC'                                       │ undefined                                    │
│ 1       │ 'ancient8' │        │             │ 'TokenNameMismatch'     │         │ 'USD Coin'                                   │ undefined                                    │
│ 2       │ 'ancient8' │        │             │ 'TokenDecimalsMismatch' │         │ 6                                            │ undefined                                    │
│ 3       │ 'ancient8' │        │ 'synthetic' │ 'ProxyAdmin'            │         │ '0xdB670e1a1e312BF17425b08cE55Bdf2cD8F8eD54' │ '0x0761b0827849abbf7b0cC09CE14e1C93D87f5004' │
└─────────┴────────────┴────────┴─────────────┴─────────────────────────┴─────────┴──────────────────────────────────────────────┴──────────────────────────────────────────────┘
```

#### Ethereum

```
$yarn tsx ./scripts/check-deploy.ts -e mainnet3 -m warp -c ethereum

Deploying staticAggregationIsm to ethereum 
Deploying messageIdMultisigIsm to ethereum 
Recovered 2 of 3 address set on ethereum: 0xD32513263Eb3190c4DA126fe418864904D9F4A32
Deploying merkleRootMultisigIsm to ethereum 
Recovered 2 of 3 address set on ethereum: 0x054B75b89DA7a428C2BF1f6b5df33944167B785C
Recovered 1 of 2 address set on ethereum: 0x9c764dF6bdd6B716CfadEAa348E1dB146694F67F
┌─────────┬────────────┬────────┬──────────────┬──────────────┬─────────┬──────────────────────────────────────────────┬──────────────────────────────────────────────┐
│ (index) │ chain      │ remote │ name         │ type         │ subType │ actual                                       │ expected                                     │
├─────────┼────────────┼────────┼──────────────┼──────────────┼─────────┼──────────────────────────────────────────────┼──────────────────────────────────────────────┤
│ 0       │ 'ethereum' │        │ 'collateral' │ 'ProxyAdmin' │         │ '0xEd96482beA3C51a33b4c1aDa8b438e33A2367413' │ '0x75EE15Ee1B4A75Fa3e2fDF5DF3253c25599cc659' │
└─────────┴────────────┴────────┴──────────────┴──────────────┴─────────┴──────────────────────────────────────────────┴──────────────────────────────────────────────┘
```

### CODE REVIEW

File: typescript/sdk/src/token/checker.ts [hunk 1]
```diff
@@ -25,6 +25,10 @@ export class HypERC20Checker extends HyperlaneRouterChecker<
   async checkChain(chain: ChainName): Promise<void> {
     await super.checkChain(chain);
     await this.checkToken(chain);
+    // We have adapted thi method to accept a proxyAdmin contract address parameter
```
Comment (tkporter) [inline typescript/sdk/src/token/checker.ts:28]:
```suggestion
    // We have adapted this method to accept a proxyAdmin contract address parameter
```
Could you give a quick read of the comment's grammar too, it made it a little hard to understand quickly

**Review by tkporter — APPROVED** [review]

File: typescript/sdk/src/router/schemas.ts [hunk 2]
```diff
@@ -24,13 +24,19 @@ export const RemoteRoutersSchema = z.record(
   RemoteRouterRouter,
 );
 
+const ProxyAdminConfigSchema = z.object({
+  proxyAdmin: z.string().optional(),
```
Comment (yorhodes) [inline typescript/sdk/src/router/schemas.ts:28]: this is also an artifact of the warp route deployment so this is subtlely mixing config and artifact schemas

File: typescript/sdk/src/token/checker.ts [hunk 3]
```diff
@@ -25,6 +25,10 @@ export class HypERC20Checker extends HyperlaneRouterChecker<
...
+    // extending HypERC20Checker class with ProxiedRouterChecker, which could be done to avoid this change
+    // leads further issues as the ProxyAdmin contract does not conform to GasRouter interface
+    await this.checkProxiedContracts(chain, this.configMap[chain].proxyAdmin);
```
Comment (yorhodes) [inline typescript/sdk/src/token/checker.ts:31]: I dont really understand this
> ProxyAdmin contract does not conform to GasRouter interface

File: typescript/sdk/src/router/schemas.ts (hunk 2 above)
Comment (Mo-Hussain) [reply typescript/sdk/src/router/schemas.ts:28]: The `RouterConfig` type is used for an artifact of the warp route deployment?

File: typescript/sdk/src/token/checker.ts (hunk 3 above)
Comment (Mo-Hussain) [reply typescript/sdk/src/token/checker.ts:31]: See my latest commit, it's my rough attempt to have `HypERC20Checker` extend `ProxiedRouterChecker`.
 
```
src/token/deploy.ts(200,9): error TS2322: Type 'TimelockController | ProxyAdmin | HypERC20 | HypERC20Collateral | HypXERC20Lockbox | HypXERC20 | ... 5 more ... | HypNativeScaled' is not assignable to type 'GasRouter'.
  Type 'TimelockController' is missing the following properties from type 'GasRouter': destinationGas, "destinationGas(uint32)", domains, "domains()", and 34 more.

15:39:19 - Found 1 error. Watching for file changes.
```

I don't think it make sense to add the `proxiedFactories` to `HypERC20Factories` anyways

Comment (yorhodes) [reply typescript/sdk/src/token/checker.ts:31]: ah I see
yes agreed we should not add `proxiedFactories` to `HypERC20Factories`
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/1ec61debd2425785ca4ade111ad3cdaba8ccc8ac/typescript/sdk/src/router/ProxiedRouterDeployer.ts#L25-L35
here is what I did in the deployer

Comment (Mo-Hussain) [reply typescript/sdk/src/token/checker.ts:31]: This approach still leads to 

```
src/token/deploy.ts(200,9): error TS2322: Type 'TimelockController | ProxyAdmin | HypERC20 | HypERC20Collateral | HypXERC20Lockbox | HypXERC20 | ... 5 more ... | HypNativeScaled' is not assignable to type 'GasRouter'.
  Type 'TimelockController' is missing the following properties from type 'GasRouter': destinationGas, "destinationGas(uint32)", domains, "domains()", and 34 more.

15:39:19 - Found 1 error. Watching for file changes.
```

I think having HypERC20Checker extend ProxiedRouterChecker will be challenging

**Pull request merged**